<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.1.175">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">

<meta name="dcterms.date" content="2022-11-04">

<title>Bottomly analysis: E-Values as Multiple Testing Weights</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1.6em;
  vertical-align: middle;
}
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { color: #008000; } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { color: #008000; font-weight: bold; } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>


<script src="bottomly_files/libs/clipboard/clipboard.min.js"></script>
<script src="bottomly_files/libs/quarto-html/quarto.js"></script>
<script src="bottomly_files/libs/quarto-html/popper.min.js"></script>
<script src="bottomly_files/libs/quarto-html/tippy.umd.min.js"></script>
<script src="bottomly_files/libs/quarto-html/anchor.min.js"></script>
<link href="bottomly_files/libs/quarto-html/tippy.css" rel="stylesheet">
<link href="bottomly_files/libs/quarto-html/quarto-syntax-highlighting.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="bottomly_files/libs/bootstrap/bootstrap.min.js"></script>
<link href="bottomly_files/libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="bottomly_files/libs/bootstrap/bootstrap.min.css" rel="stylesheet" id="quarto-bootstrap" data-mode="light">

<script src="bottomly_files/libs/kePrint-0.0.1/kePrint.js"></script>
<link href="bottomly_files/libs/lightable-0.0.1/lightable.css" rel="stylesheet">


</head>

<body class="fullcontent">

<div id="quarto-content" class="page-columns page-rows-contents page-layout-article">

<main class="content" id="quarto-document-content">

<header id="title-block-header" class="quarto-title-block default">
<div class="quarto-title">
<h1 class="title">Bottomly analysis: E-Values as Multiple Testing Weights</h1>
</div>



<div class="quarto-title-meta">

    
    <div>
    <div class="quarto-title-meta-heading">Published</div>
    <div class="quarto-title-meta-contents">
      <p class="date">November 4, 2022</p>
    </div>
  </div>
    
  </div>
  

</header>

<div class="cell">
<div class="sourceCode cell-code" id="cb1"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(here)</span>
<span id="cb1-2"><a href="#cb1-2" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(sessioninfo)</span>
<span id="cb1-3"><a href="#cb1-3" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(EValueWeightsPaper)</span>
<span id="cb1-4"><a href="#cb1-4" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(DESeq2)</span>
<span id="cb1-5"><a href="#cb1-5" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(tidyverse)</span>
<span id="cb1-6"><a href="#cb1-6" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(GEOquery)</span>
<span id="cb1-7"><a href="#cb1-7" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(utils)</span>
<span id="cb1-8"><a href="#cb1-8" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(mouse4302.db)</span>
<span id="cb1-9"><a href="#cb1-9" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(AnnotationDbi)</span>
<span id="cb1-10"><a href="#cb1-10" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(limma)</span>
<span id="cb1-11"><a href="#cb1-11" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(oligo)</span>
<span id="cb1-12"><a href="#cb1-12" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(cowplot)</span>
<span id="cb1-13"><a href="#cb1-13" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(IHW)</span>
<span id="cb1-14"><a href="#cb1-14" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(kableExtra)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<section id="deseq2-analysis" class="level1">
<h1>DESeq2 analysis</h1>
<p>Load dataset:</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb2"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb2-1"><a href="#cb2-1" aria-hidden="true" tabindex="-1"></a><span class="fu">data</span>(<span class="st">"bottomly"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>Subset to the first two batches:</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb3"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb3-1"><a href="#cb3-1" aria-hidden="true" tabindex="-1"></a>bottomly <span class="ot">&lt;-</span> bottomly[, bottomly<span class="sc">$</span>batch <span class="sc">%in%</span> <span class="fu">c</span>(<span class="st">"b4"</span>, <span class="st">"b6"</span>)]</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb4"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb4-1"><a href="#cb4-1" aria-hidden="true" tabindex="-1"></a><span class="fu">colData</span>(bottomly)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>DataFrame with 14 rows and 11 columns
          experiment sample.id num.tech.reps   strain experiment.number
            &lt;factor&gt;  &lt;factor&gt;     &lt;integer&gt; &lt;factor&gt;         &lt;integer&gt;
SRR099223  SRX033472 SRX033472             1 DBA.2J                   4
SRR099224  SRX033473 SRX033473             1 DBA.2J                   4
SRR099225  SRX033474 SRX033474             1 DBA.2J                   4
SRR099226  SRX033475 SRX033475             1 DBA.2J                   4
SRR099227  SRX033476 SRX033476             1 C57BL.6J                 4
...              ...       ...           ...      ...               ...
SRR099232  SRX033482 SRX033482             1 C57BL.6J                 6
SRR099233  SRX033483 SRX033483             1 C57BL.6J                 6
SRR099234  SRX033484 SRX033484             1 DBA.2J                   6
SRR099235  SRX033485 SRX033485             1 DBA.2J                   6
SRR099236  SRX033486 SRX033486             1 DBA.2J                   6
          lane.number submission     study    sample         run    batch
            &lt;integer&gt;   &lt;factor&gt;  &lt;factor&gt;  &lt;factor&gt; &lt;character&gt; &lt;factor&gt;
SRR099223           1  SRA026846 SRP004777 SRS140384   SRR099223       b4
SRR099224           2  SRA026846 SRP004777 SRS140385   SRR099224       b4
SRR099225           3  SRA026846 SRP004777 SRS140386   SRR099225       b4
SRR099226           5  SRA026846 SRP004777 SRS140387   SRR099226       b4
SRR099227           6  SRA026846 SRP004777 SRS140388   SRR099227       b4
...               ...        ...       ...       ...         ...      ...
SRR099232           3  SRA026846 SRP004777 SRS140393   SRR099232       b6
SRR099233           5  SRA026846 SRP004777 SRS140394   SRR099233       b6
SRR099234           6  SRA026846 SRP004777 SRS140395   SRR099234       b6
SRR099235           7  SRA026846 SRP004777 SRS140396   SRR099235       b6
SRR099236           8  SRA026846 SRP004777 SRS140397   SRR099236       b6</code></pre>
</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb6"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb6-1"><a href="#cb6-1" aria-hidden="true" tabindex="-1"></a><span class="fu">table</span>(bottomly<span class="sc">$</span>strain)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>
C57BL.6J   DBA.2J 
       7        7 </code></pre>
</div>
</div>
<p>Run DESeq2 analysis.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb8"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb8-1"><a href="#cb8-1" aria-hidden="true" tabindex="-1"></a>dds <span class="ot">&lt;-</span> <span class="fu">DESeqDataSet</span>(<span class="at">se=</span>bottomly, <span class="at">design=</span> <span class="sc">~</span> batch <span class="sc">+</span> strain)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>factor levels were dropped which had no samples</code></pre>
</div>
<div class="sourceCode cell-code" id="cb10"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb10-1"><a href="#cb10-1" aria-hidden="true" tabindex="-1"></a>dds <span class="ot">&lt;-</span> <span class="fu">DESeq</span>(dds)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>estimating size factors</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>estimating dispersions</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>gene-wise dispersion estimates</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>mean-dispersion relationship</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>final dispersion estimates</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>fitting model and testing</code></pre>
</div>
<div class="sourceCode cell-code" id="cb17"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb17-1"><a href="#cb17-1" aria-hidden="true" tabindex="-1"></a>res_bottomly <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(<span class="fu">results</span>(dds)) <span class="sc">%&gt;%</span> <span class="fu">filter</span>(baseMean <span class="sc">&gt;=</span> <span class="fl">0.2</span>, <span class="sc">!</span><span class="fu">is.na</span>(pvalue))</span>
<span id="cb17-2"><a href="#cb17-2" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(res_bottomly)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>                     baseMean log2FoldChange      lfcSE        stat    pvalue
ENSMUSG00000000001 408.555049    -0.11351754 0.10791573 -1.05190914 0.2928412
ENSMUSG00000000028  13.467412    -0.07819158 0.37633357 -0.20777199 0.8354070
ENSMUSG00000000031   0.760055     0.50231765 1.39620975  0.35977234 0.7190174
ENSMUSG00000000037  21.872988    -0.01464436 0.43799875 -0.03343471 0.9733279
ENSMUSG00000000049   1.261782    -0.11585165 1.04519851 -0.11084177 0.9117418
ENSMUSG00000000056 775.289329    -0.04348897 0.07716155 -0.56360928 0.5730201
                        padj
ENSMUSG00000000001 0.5467497
ENSMUSG00000000028 0.9289843
ENSMUSG00000000031        NA
ENSMUSG00000000037 0.9897943
ENSMUSG00000000049 0.9646881
ENSMUSG00000000056 0.7785793</code></pre>
</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb19"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb19-1"><a href="#cb19-1" aria-hidden="true" tabindex="-1"></a><span class="fu">nrow</span>(res_bottomly)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] 24906</code></pre>
</div>
</div>
</section>
<section id="microarray-analysis" class="level1">
<h1>Microarray analysis</h1>
<section id="download-files" class="level2">
<h2 class="anchored" data-anchor-id="download-files">Download files</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb21"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb21-1"><a href="#cb21-1" aria-hidden="true" tabindex="-1"></a>files_dir <span class="ot">&lt;-</span> <span class="fu">here</span>(<span class="st">"data_analysis"</span>,<span class="st">"microarray_files"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb22"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb22-1"><a href="#cb22-1" aria-hidden="true" tabindex="-1"></a>microarray <span class="ot">&lt;-</span> <span class="fu">getGEO</span>(<span class="at">GEO=</span><span class="st">"GSE26024"</span>, <span class="at">destdir=</span>files_dir, <span class="at">GSEMatrix=</span><span class="cn">TRUE</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Found 2 file(s)</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>GSE26024-GPL1261_series_matrix.txt.gz</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Using locally cached version: /Users/nikos/Documents/Stanford/Aadi/EValueWeightsPaper/data_analysis/microarray_files/GSE26024-GPL1261_series_matrix.txt.gz</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Using locally cached version of GPL1261 found here:
/Users/nikos/Documents/Stanford/Aadi/EValueWeightsPaper/data_analysis/microarray_files/GPL1261.soft.gz </code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>GSE26024-GPL6885_series_matrix.txt.gz</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Using locally cached version: /Users/nikos/Documents/Stanford/Aadi/EValueWeightsPaper/data_analysis/microarray_files/GSE26024-GPL6885_series_matrix.txt.gz</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Using locally cached version of GPL6885 found here:
/Users/nikos/Documents/Stanford/Aadi/EValueWeightsPaper/data_analysis/microarray_files/GPL6885.soft.gz </code></pre>
</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb30"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb30-1"><a href="#cb30-1" aria-hidden="true" tabindex="-1"></a>suppl_file_names <span class="ot">&lt;-</span> <span class="fu">list.files</span>(<span class="fu">file.path</span>(files_dir, <span class="st">"GSE26024_RAW"</span>))</span>
<span id="cb30-2"><a href="#cb30-2" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span> (<span class="fu">length</span>(suppl_file_names) <span class="sc">==</span> <span class="dv">0</span>){</span>
<span id="cb30-3"><a href="#cb30-3" aria-hidden="true" tabindex="-1"></a>  microarray_suppl <span class="ot">&lt;-</span> <span class="fu">getGEOSuppFiles</span>(<span class="at">GEO=</span><span class="st">"GSE26024"</span>, <span class="at">baseDir=</span>files_dir, <span class="at">makeDirectory=</span><span class="cn">FALSE</span>)</span>
<span id="cb30-4"><a href="#cb30-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">untar</span>(<span class="fu">file.path</span>(files_dir, <span class="st">"GSE26024_RAW.tar"</span>),</span>
<span id="cb30-5"><a href="#cb30-5" aria-hidden="true" tabindex="-1"></a>      <span class="at">exdir=</span><span class="fu">file.path</span>(files_dir, <span class="st">"GSE26024_RAW"</span>))</span>
<span id="cb30-6"><a href="#cb30-6" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb30-7"><a href="#cb30-7" aria-hidden="true" tabindex="-1"></a>suppl_file_names <span class="ot">&lt;-</span> <span class="fu">list.files</span>(<span class="fu">file.path</span>(files_dir, <span class="st">"GSE26024_RAW"</span>))[<span class="dv">2</span><span class="sc">:</span><span class="dv">21</span>]</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="preprocessing-and-normalization" class="level2">
<h2 class="anchored" data-anchor-id="preprocessing-and-normalization">Preprocessing and normalization</h2>
<p>Prepare metadata for microarray.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb31"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb31-1"><a href="#cb31-1" aria-hidden="true" tabindex="-1"></a>affy <span class="ot">&lt;-</span> microarray[[<span class="dv">1</span>]]</span>
<span id="cb31-2"><a href="#cb31-2" aria-hidden="true" tabindex="-1"></a>affy_pdata <span class="ot">&lt;-</span> <span class="fu">pData</span>(affy)</span>
<span id="cb31-3"><a href="#cb31-3" aria-hidden="true" tabindex="-1"></a>affy_fdata <span class="ot">&lt;-</span> <span class="fu">featureData</span>(affy)</span>
<span id="cb31-4"><a href="#cb31-4" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(affy_pdata) <span class="ot">&lt;-</span> suppl_file_names</span>
<span id="cb31-5"><a href="#cb31-5" aria-hidden="true" tabindex="-1"></a><span class="co"># sanity check</span></span>
<span id="cb31-6"><a href="#cb31-6" aria-hidden="true" tabindex="-1"></a><span class="fu">all</span>(<span class="fu">setNames</span>(<span class="fu">sapply</span>(affy_pdata<span class="sc">$</span>supplementary_file, <span class="cf">function</span>(x) <span class="fu">last</span>(<span class="fu">strsplit</span>(x, <span class="st">"/"</span>)[[<span class="dv">1</span>]])), <span class="cn">NULL</span>) <span class="sc">==</span> <span class="fu">rownames</span>(affy_pdata))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] TRUE</code></pre>
</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb33"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb33-1"><a href="#cb33-1" aria-hidden="true" tabindex="-1"></a>affy_pdata <span class="ot">&lt;-</span> <span class="fu">AnnotatedDataFrame</span>(affy_pdata)</span>
<span id="cb33-2"><a href="#cb33-2" aria-hidden="true" tabindex="-1"></a>affy_pdata<span class="sc">$</span>gender <span class="ot">&lt;-</span> <span class="fu">factor</span>(<span class="fu">ifelse</span>(affy_pdata<span class="sc">$</span>characteristics_ch1 <span class="sc">==</span> <span class="st">"gender: male"</span>, <span class="st">"male"</span>, <span class="st">"female"</span>))</span>
<span id="cb33-3"><a href="#cb33-3" aria-hidden="true" tabindex="-1"></a>affy_pdata<span class="sc">$</span>strain <span class="ot">&lt;-</span>  <span class="fu">factor</span>(<span class="fu">ifelse</span>(affy_pdata<span class="sc">$</span>characteristics_ch1<span class="fl">.1</span> <span class="sc">==</span> <span class="st">"strain: C57BL/6J"</span>, <span class="st">"C57BL"</span>, <span class="st">"DBA"</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>Load in raw data and filter to male mice.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb34"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb34-1"><a href="#cb34-1" aria-hidden="true" tabindex="-1"></a>affy_raw_data <span class="ot">&lt;-</span> oligo<span class="sc">::</span><span class="fu">read.celfiles</span>(<span class="at">filenames =</span> <span class="fu">file.path</span>(files_dir, <span class="st">"GSE26024_RAW"</span>, suppl_file_names),</span>
<span id="cb34-2"><a href="#cb34-2" aria-hidden="true" tabindex="-1"></a>                                 <span class="at">verbose =</span> <span class="cn">FALSE</span>, <span class="at">phenoData =</span> affy_pdata)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in oligo::read.celfiles(filenames = file.path(files_dir,
"GSE26024_RAW", : 'channel' automatically added to varMetadata in phenoData.</code></pre>
</div>
<div class="sourceCode cell-code" id="cb36"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb36-1"><a href="#cb36-1" aria-hidden="true" tabindex="-1"></a>affy_raw_data <span class="ot">&lt;-</span> affy_raw_data[, affy_raw_data<span class="sc">$</span>gender<span class="sc">==</span><span class="st">"male"</span>]</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>Run RMA normalization.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb37"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb37-1"><a href="#cb37-1" aria-hidden="true" tabindex="-1"></a>rma_affy <span class="ot">&lt;-</span> oligo<span class="sc">::</span><span class="fu">rma</span>(affy_raw_data)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>Background correcting
Normalizing
Calculating Expression</code></pre>
</div>
</div>
</section>
<section id="match-probe-ids-to-ensembl-ids-to-be-able-to-match-microarray-based-tests-to-the-rnaseq-ones" class="level2">
<h2 class="anchored" data-anchor-id="match-probe-ids-to-ensembl-ids-to-be-able-to-match-microarray-based-tests-to-the-rnaseq-ones">Match Probe IDs to ENSEMBL ids (to be able to match microarray-based tests to the RNASeq ones)</h2>
<p>Construct 1 to many mapping from probe ids to ensembl ids.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb39"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb39-1"><a href="#cb39-1" aria-hidden="true" tabindex="-1"></a>anno_bottomly <span class="ot">&lt;-</span> AnnotationDbi<span class="sc">::</span><span class="fu">select</span>(mouse4302.db,</span>
<span id="cb39-2"><a href="#cb39-2" aria-hidden="true" tabindex="-1"></a>                                       <span class="at">keys =</span> (<span class="fu">featureNames</span>(rma_affy)),</span>
<span id="cb39-3"><a href="#cb39-3" aria-hidden="true" tabindex="-1"></a>                                       <span class="at">columns =</span> <span class="fu">c</span>(<span class="st">"ENSEMBL"</span>),</span>
<span id="cb39-4"><a href="#cb39-4" aria-hidden="true" tabindex="-1"></a>                                       <span class="at">keytype =</span> <span class="st">"PROBEID"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>'select()' returned 1:many mapping between keys and columns</code></pre>
</div>
</div>
<p>Remove probes that either did not match to any ensembl id, or that matched to multiple ids.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb41"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb41-1"><a href="#cb41-1" aria-hidden="true" tabindex="-1"></a>anno_bottomly_no_of_matches <span class="ot">&lt;-</span> <span class="fu">group_by</span>(anno_bottomly, PROBEID) <span class="sc">%&gt;%</span></span>
<span id="cb41-2"><a href="#cb41-2" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">summarize</span>(<span class="at">is_na =</span> <span class="fu">any</span>(<span class="fu">is.na</span>(ENSEMBL)),</span>
<span id="cb41-3"><a href="#cb41-3" aria-hidden="true" tabindex="-1"></a>                     <span class="at">no_of_matches =</span> <span class="fu">n_distinct</span>(ENSEMBL))</span>
<span id="cb41-4"><a href="#cb41-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb41-5"><a href="#cb41-5" aria-hidden="true" tabindex="-1"></a>filtered_ids <span class="ot">&lt;-</span> (anno_bottomly_no_of_matches<span class="sc">$</span>no_of_matches <span class="sc">==</span> <span class="dv">1</span>) <span class="sc">&amp;</span></span>
<span id="cb41-6"><a href="#cb41-6" aria-hidden="true" tabindex="-1"></a>                (<span class="sc">!</span>anno_bottomly_no_of_matches<span class="sc">$</span>is_na)</span>
<span id="cb41-7"><a href="#cb41-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb41-8"><a href="#cb41-8" aria-hidden="true" tabindex="-1"></a>rma_affy_filt <span class="ot">&lt;-</span> rma_affy[anno_bottomly_no_of_matches<span class="sc">$</span>PROBEID[filtered_ids], ]</span>
<span id="cb41-9"><a href="#cb41-9" aria-hidden="true" tabindex="-1"></a><span class="fu">stopifnot</span>(<span class="fu">all</span>(<span class="fu">rownames</span>(rma_affy[filtered_ids, ]) <span class="sc">==</span> <span class="fu">rownames</span>(rma_affy_filt)))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>Get the 1 to 1 matching of probe ids to ENSEMBL ids for filtered subset.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb42"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb42-1"><a href="#cb42-1" aria-hidden="true" tabindex="-1"></a>anno_bottomly_filt <span class="ot">&lt;-</span> AnnotationDbi<span class="sc">::</span><span class="fu">select</span>(mouse4302.db,</span>
<span id="cb42-2"><a href="#cb42-2" aria-hidden="true" tabindex="-1"></a>                                       <span class="at">keys =</span> (<span class="fu">featureNames</span>(rma_affy_filt)),</span>
<span id="cb42-3"><a href="#cb42-3" aria-hidden="true" tabindex="-1"></a>                                       <span class="at">columns =</span> <span class="fu">c</span>(<span class="st">"ENSEMBL"</span>),</span>
<span id="cb42-4"><a href="#cb42-4" aria-hidden="true" tabindex="-1"></a>                                       <span class="at">keytype =</span> <span class="st">"PROBEID"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>'select()' returned 1:1 mapping between keys and columns</code></pre>
</div>
<div class="sourceCode cell-code" id="cb44"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb44-1"><a href="#cb44-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Sanity check</span></span>
<span id="cb44-2"><a href="#cb44-2" aria-hidden="true" tabindex="-1"></a><span class="fu">stopifnot</span>(<span class="fu">all</span>(<span class="fu">rownames</span>(rma_affy_filt) <span class="sc">==</span> anno_bottomly_filt<span class="sc">$</span>PROBEID))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>When multiple probes match to the same ENSEMBL id, only keep the one with the largest average intensity (as recommended by Gordon Smyth in the <a href="https://support.bioconductor.org/p/126306/">Bioconductor support forum</a>).</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb45"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb45-1"><a href="#cb45-1" aria-hidden="true" tabindex="-1"></a>A <span class="ot">&lt;-</span> <span class="fu">rowMeans</span>(<span class="fu">exprs</span>(rma_affy_filt))</span>
<span id="cb45-2"><a href="#cb45-2" aria-hidden="true" tabindex="-1"></a>o <span class="ot">&lt;-</span> <span class="fu">order</span>(A, <span class="at">decreasing=</span><span class="cn">TRUE</span>)</span>
<span id="cb45-3"><a href="#cb45-3" aria-hidden="true" tabindex="-1"></a>ordered_ensembl_id <span class="ot">&lt;-</span> anno_bottomly_filt<span class="sc">$</span>ENSEMBL[o]</span>
<span id="cb45-4"><a href="#cb45-4" aria-hidden="true" tabindex="-1"></a>d <span class="ot">&lt;-</span> <span class="fu">duplicated</span>(ordered_ensembl_id)</span>
<span id="cb45-5"><a href="#cb45-5" aria-hidden="true" tabindex="-1"></a>o_filt <span class="ot">&lt;-</span> o[<span class="sc">!</span>d]</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb46"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb46-1"><a href="#cb46-1" aria-hidden="true" tabindex="-1"></a>rma_affy_gene <span class="ot">&lt;-</span> rma_affy_filt[o_filt, ]</span>
<span id="cb46-2"><a href="#cb46-2" aria-hidden="true" tabindex="-1"></a><span class="fu">stopifnot</span>(<span class="fu">all</span>(anno_bottomly_filt<span class="sc">$</span>PROBEID[o_filt] <span class="sc">==</span> <span class="fu">rownames</span>(rma_affy_gene)))</span>
<span id="cb46-3"><a href="#cb46-3" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(rma_affy_gene) <span class="ot">&lt;-</span> anno_bottomly_filt<span class="sc">$</span>ENSEMBL[o_filt]</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="limma-analysis" class="level2">
<h2 class="anchored" data-anchor-id="limma-analysis">LIMMA analysis</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb47"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb47-1"><a href="#cb47-1" aria-hidden="true" tabindex="-1"></a>affy_design <span class="ot">&lt;-</span> <span class="fu">with</span>(<span class="fu">pData</span>(rma_affy_gene), <span class="fu">model.matrix</span>( <span class="sc">~</span> strain))</span>
<span id="cb47-2"><a href="#cb47-2" aria-hidden="true" tabindex="-1"></a>affy_design</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>   (Intercept) strainDBA
1            1         0
2            1         0
3            1         0
4            1         0
5            1         0
6            1         1
7            1         1
8            1         1
9            1         1
10           1         1
attr(,"assign")
[1] 0 1
attr(,"contrasts")
attr(,"contrasts")$strain
[1] "contr.treatment"</code></pre>
</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb49"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb49-1"><a href="#cb49-1" aria-hidden="true" tabindex="-1"></a>affy_lm_fit <span class="ot">&lt;-</span> <span class="fu">lmFit</span>(rma_affy_gene, affy_design)</span>
<span id="cb49-2"><a href="#cb49-2" aria-hidden="true" tabindex="-1"></a>affy_ebayes_lm_fit <span class="ot">&lt;-</span> <span class="fu">eBayes</span>(affy_lm_fit)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>Parameter estimates of limma model:</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb50"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb50-1"><a href="#cb50-1" aria-hidden="true" tabindex="-1"></a><span class="fu">c</span>(affy_ebayes_lm_fit<span class="sc">$</span>df.prior, affy_ebayes_lm_fit<span class="sc">$</span>s2.prior)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] 3.63576081 0.01439132</code></pre>
</div>
</div>
<p>The helper function below also computes the e-values.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb52"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb52-1"><a href="#cb52-1" aria-hidden="true" tabindex="-1"></a>affy_res <span class="ot">&lt;-</span> <span class="fu">limma_results_table_with_evalues</span>(affy_ebayes_lm_fit, <span class="at">coef=</span><span class="st">"strainDBA"</span>)</span>
<span id="cb52-2"><a href="#cb52-2" aria-hidden="true" tabindex="-1"></a><span class="fu">stopifnot</span>( <span class="fu">max</span>(<span class="fu">abs</span>(<span class="fu">log</span>(affy_res<span class="sc">$</span>evalue) <span class="sc">-</span> <span class="fu">eBayes</span>(affy_lm_fit,<span class="at">proportion=</span><span class="fl">0.5</span>)<span class="sc">$</span>lods[,<span class="dv">2</span>])) <span class="sc">&lt;</span> <span class="fl">1e-12</span>)</span>
<span id="cb52-3"><a href="#cb52-3" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(affy_res)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>                            t   P.Value         B data_dof    evalue
ENSMUSG00000036438 -0.2739394 0.7889334 -7.141720        8 0.4804309
ENSMUSG00000041607 -0.4598836 0.6540741 -7.068952        8 0.5087044
ENSMUSG00000034892  0.3924394 0.7018273 -7.099505        8 0.4966447
ENSMUSG00000071866  0.3264527 0.7498787 -7.124831        8 0.4868559
ENSMUSG00000037742 -0.5643877 0.5832062 -7.012455        8 0.5317464
ENSMUSG00000036564  0.1890498 0.8533111 -7.162839        8 0.4725112</code></pre>
</div>
</div>
</section>
</section>
<section id="combined-analysis" class="level1">
<h1>Combined analysis</h1>
<section id="joint-data-frame" class="level2">
<h2 class="anchored" data-anchor-id="joint-data-frame">Joint data frame</h2>
<p>Map each row of the RNASeq dataset to rows of the microarray dataset (return NA if there is no match).</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb54"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb54-1"><a href="#cb54-1" aria-hidden="true" tabindex="-1"></a>match_idx <span class="ot">&lt;-</span> <span class="fu">match</span>(<span class="fu">rownames</span>(res_bottomly), <span class="fu">rownames</span>(affy_res))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb55"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb55-1"><a href="#cb55-1" aria-hidden="true" tabindex="-1"></a><span class="fu">c</span>(<span class="fu">sum</span>(<span class="sc">!</span><span class="fu">is.na</span>(match_idx)), <span class="fu">sum</span>(<span class="fu">is.na</span>(match_idx)))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] 15870  9036</code></pre>
</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb57"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb57-1"><a href="#cb57-1" aria-hidden="true" tabindex="-1"></a><span class="fu">c</span>(<span class="fu">rownames</span>(res_bottomly)[<span class="dv">1</span><span class="sc">:</span><span class="dv">3</span>], <span class="fu">rownames</span>(affy_res)[match_idx[<span class="dv">1</span><span class="sc">:</span><span class="dv">3</span>]])</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] "ENSMUSG00000000001" "ENSMUSG00000000028" "ENSMUSG00000000031"
[4] "ENSMUSG00000000001" "ENSMUSG00000000028" "ENSMUSG00000000031"</code></pre>
</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb59"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb59-1"><a href="#cb59-1" aria-hidden="true" tabindex="-1"></a>microarray_pvalues <span class="ot">&lt;-</span> affy_res<span class="sc">$</span>P.Value[match_idx]</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>Set e-value of genes that did not have a match in the microarray dataset to 1:</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb60"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb60-1"><a href="#cb60-1" aria-hidden="true" tabindex="-1"></a>microarray_evalues <span class="ot">&lt;-</span> affy_res<span class="sc">$</span>evalue[match_idx]</span>
<span id="cb60-2"><a href="#cb60-2" aria-hidden="true" tabindex="-1"></a>microarray_evalues[<span class="fu">is.na</span>(microarray_evalues)] <span class="ot">&lt;-</span> <span class="dv">1</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb61"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb61-1"><a href="#cb61-1" aria-hidden="true" tabindex="-1"></a>res <span class="ot">&lt;-</span> res_bottomly <span class="sc">%&gt;%</span></span>
<span id="cb61-2"><a href="#cb61-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">microarray_pvalue =</span> microarray_pvalues,</span>
<span id="cb61-3"><a href="#cb61-3" aria-hidden="true" tabindex="-1"></a>        <span class="at">evalue=</span>microarray_evalues,</span>
<span id="cb61-4"><a href="#cb61-4" aria-hidden="true" tabindex="-1"></a>        <span class="at">pe =</span> <span class="fu">pmin</span>(pvalue<span class="sc">/</span> evalue, <span class="dv">1</span>))</span>
<span id="cb61-5"><a href="#cb61-5" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(res)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>                     baseMean log2FoldChange      lfcSE        stat    pvalue
ENSMUSG00000000001 408.555049    -0.11351754 0.10791573 -1.05190914 0.2928412
ENSMUSG00000000028  13.467412    -0.07819158 0.37633357 -0.20777199 0.8354070
ENSMUSG00000000031   0.760055     0.50231765 1.39620975  0.35977234 0.7190174
ENSMUSG00000000037  21.872988    -0.01464436 0.43799875 -0.03343471 0.9733279
ENSMUSG00000000049   1.261782    -0.11585165 1.04519851 -0.11084177 0.9117418
ENSMUSG00000000056 775.289329    -0.04348897 0.07716155 -0.56360928 0.5730201
                        padj microarray_pvalue    evalue        pe
ENSMUSG00000000001 0.5467497         0.5450349 0.5472602 0.5351042
ENSMUSG00000000028 0.9289843         0.3129606 0.7279727 1.0000000
ENSMUSG00000000031        NA         0.3739063 0.6592644 1.0000000
ENSMUSG00000000037 0.9897943         0.4655961 0.5887562 1.0000000
ENSMUSG00000000049 0.9646881         0.4364047 0.6080067 1.0000000
ENSMUSG00000000056 0.7785793         0.1170802 1.3467318 0.4254894</code></pre>
</div>
</div>
</section>
<section id="plot-p-values-e-values-and-merged-pe-values" class="level2">
<h2 class="anchored" data-anchor-id="plot-p-values-e-values-and-merged-pe-values">Plot p-values, e-values and merged pe-values</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb63"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb63-1"><a href="#cb63-1" aria-hidden="true" tabindex="-1"></a>new_df <span class="ot">&lt;-</span> <span class="fu">rbind</span>( <span class="fu">data.frame</span>( <span class="at">pvalue =</span> res<span class="sc">$</span>pvalue, <span class="at">group=</span><span class="st">"vanilla"</span>),</span>
<span id="cb63-2"><a href="#cb63-2" aria-hidden="true" tabindex="-1"></a>                 <span class="fu">data.frame</span>( <span class="at">pvalue =</span> res<span class="sc">$</span>pe, <span class="at">group=</span><span class="st">"P/E"</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb63-3"><a href="#cb63-3" aria-hidden="true" tabindex="-1"></a>          <span class="fu">mutate</span>(<span class="at">group =</span> <span class="fu">factor</span>(group, <span class="at">levels=</span><span class="fu">c</span>(<span class="st">"vanilla"</span>,<span class="st">"P/E"</span>)))</span>
<span id="cb63-4"><a href="#cb63-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb63-5"><a href="#cb63-5" aria-hidden="true" tabindex="-1"></a>hist_pvalue <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(res, <span class="fu">aes</span>(<span class="at">x =</span> pvalue)) <span class="sc">+</span></span>
<span id="cb63-6"><a href="#cb63-6" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_histogram</span>(<span class="at">binwidth =</span> <span class="fl">0.025</span>, <span class="at">boundary =</span> <span class="dv">0</span>, <span class="at">alpha=</span><span class="fl">0.4</span>, <span class="at">color=</span><span class="st">"black"</span>, <span class="at">fill=</span><span class="st">"black"</span>) <span class="sc">+</span></span>
<span id="cb63-7"><a href="#cb63-7" aria-hidden="true" tabindex="-1"></a>    <span class="fu">xlab</span>(<span class="fu">expression</span>(P[i])) <span class="sc">+</span> <span class="fu">theme_cowplot</span>()</span>
<span id="cb63-8"><a href="#cb63-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb63-9"><a href="#cb63-9" aria-hidden="true" tabindex="-1"></a>hist_evalue <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(<span class="fu">filter</span>(res, evalue <span class="sc">!=</span><span class="dv">1</span>), <span class="fu">aes</span>(<span class="at">x =</span> <span class="fu">pmin</span>(evalue,<span class="dv">5</span>))) <span class="sc">+</span></span>
<span id="cb63-10"><a href="#cb63-10" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_histogram</span>(<span class="at">boundary=</span><span class="fl">0.46</span>, <span class="at">binwidth=</span><span class="fl">0.1</span>, <span class="at">alpha=</span><span class="fl">0.4</span>, <span class="at">color=</span><span class="st">"black"</span>, <span class="at">fill=</span><span class="st">"blue"</span>) <span class="sc">+</span></span>
<span id="cb63-11"><a href="#cb63-11" aria-hidden="true" tabindex="-1"></a>    <span class="fu">xlab</span>(<span class="fu">expression</span>(<span class="st">"min("</span><span class="sc">~</span>E[i]<span class="sc">~</span><span class="st">", 5)"</span>)) <span class="sc">+</span></span>
<span id="cb63-12"><a href="#cb63-12" aria-hidden="true" tabindex="-1"></a>    <span class="fu">scale_x_continuous</span>(<span class="at">breaks=</span><span class="fu">c</span>(<span class="fl">0.46</span>, <span class="dv">1</span>, <span class="dv">2</span>, <span class="dv">3</span>, <span class="dv">4</span>, <span class="dv">5</span>)) <span class="sc">+</span></span>
<span id="cb63-13"><a href="#cb63-13" aria-hidden="true" tabindex="-1"></a>    <span class="fu">theme_cowplot</span>()</span>
<span id="cb63-14"><a href="#cb63-14" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb63-15"><a href="#cb63-15" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb63-16"><a href="#cb63-16" aria-hidden="true" tabindex="-1"></a>merged_hist <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(new_df, <span class="fu">aes</span>(<span class="at">x =</span> pvalue, <span class="at">fill =</span> group)) <span class="sc">+</span></span>
<span id="cb63-17"><a href="#cb63-17" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_histogram</span>(<span class="at">binwidth =</span> <span class="fl">0.025</span>, <span class="at">boundary =</span> <span class="dv">0</span>, <span class="at">alpha=</span><span class="fl">0.4</span>, <span class="at">color=</span><span class="st">"black"</span>, <span class="at">position=</span><span class="st">"identity"</span>) <span class="sc">+</span></span>
<span id="cb63-18"><a href="#cb63-18" aria-hidden="true" tabindex="-1"></a>    <span class="fu">theme_cowplot</span>() <span class="sc">+</span></span>
<span id="cb63-19"><a href="#cb63-19" aria-hidden="true" tabindex="-1"></a>    <span class="fu">xlab</span>(<span class="st">"statistic"</span>) <span class="sc">+</span></span>
<span id="cb63-20"><a href="#cb63-20" aria-hidden="true" tabindex="-1"></a>    <span class="fu">scale_fill_manual</span>(<span class="at">values=</span><span class="fu">c</span>(<span class="st">"black"</span>, <span class="st">"purple"</span>),</span>
<span id="cb63-21"><a href="#cb63-21" aria-hidden="true" tabindex="-1"></a>                      <span class="at">labels =</span> <span class="fu">c</span>(<span class="fu">expression</span>(P[i]),</span>
<span id="cb63-22"><a href="#cb63-22" aria-hidden="true" tabindex="-1"></a>                                 <span class="fu">expression</span>(<span class="st">"min("</span><span class="sc">~</span>P[i]<span class="sc">/</span>E[i]<span class="sc">~</span><span class="st">",1)"</span>))) <span class="sc">+</span></span>
<span id="cb63-23"><a href="#cb63-23" aria-hidden="true" tabindex="-1"></a>    <span class="fu">theme</span>(<span class="at">legend.position =</span> <span class="fu">c</span>(<span class="fl">0.3</span>, <span class="fl">0.7</span>), <span class="at">legend.title=</span><span class="fu">element_blank</span>(), <span class="at">legend.text.align =</span> <span class="dv">0</span>)</span>
<span id="cb63-24"><a href="#cb63-24" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb63-25"><a href="#cb63-25" aria-hidden="true" tabindex="-1"></a>all_panels <span class="ot">&lt;-</span> <span class="fu">plot_grid</span>(hist_pvalue, hist_evalue, merged_hist, <span class="at">labels =</span> <span class="fu">c</span>(<span class="st">'A'</span>, <span class="st">'B'</span>, <span class="st">'C'</span>), <span class="at">label_size =</span> <span class="dv">12</span>, <span class="at">nrow=</span><span class="dv">1</span>)</span>
<span id="cb63-26"><a href="#cb63-26" aria-hidden="true" tabindex="-1"></a>all_panels</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<p><img src="bottomly_files/figure-html/unnamed-chunk-30-1.png" class="img-fluid" width="1248"></p>
</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb64"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb64-1"><a href="#cb64-1" aria-hidden="true" tabindex="-1"></a><span class="fu">save_plot</span>(<span class="st">"evalue_weights.pdf"</span>, all_panels, <span class="at">base_height=</span><span class="dv">4</span>, <span class="at">base_width=</span><span class="dv">13</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="apply-bh-type-multiple-testing-methods" class="level2">
<h2 class="anchored" data-anchor-id="apply-bh-type-multiple-testing-methods">Apply BH-type multiple testing methods</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb65"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb65-1"><a href="#cb65-1" aria-hidden="true" tabindex="-1"></a>alpha_level <span class="ot">&lt;-</span> <span class="fl">0.01</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>Unweighted BH</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb66"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb66-1"><a href="#cb66-1" aria-hidden="true" tabindex="-1"></a>unweighted_bh_rjs <span class="ot">&lt;-</span><span class="fu">sum</span>(<span class="fu">p.adjust</span>(res<span class="sc">$</span>pvalue, <span class="at">method=</span><span class="st">"BH"</span>) <span class="sc">&lt;=</span> alpha_level)</span>
<span id="cb66-2"><a href="#cb66-2" aria-hidden="true" tabindex="-1"></a>unweighted_bh_90_quantile <span class="ot">&lt;-</span> <span class="dv">1</span></span>
<span id="cb66-3"><a href="#cb66-3" aria-hidden="true" tabindex="-1"></a>unweighted_bh_avg <span class="ot">&lt;-</span> <span class="dv">1</span></span>
<span id="cb66-4"><a href="#cb66-4" aria-hidden="true" tabindex="-1"></a><span class="fu">c</span>(<span class="at">rejections=</span>unweighted_bh_rjs, </span>
<span id="cb66-5"><a href="#cb66-5" aria-hidden="true" tabindex="-1"></a>  <span class="at">avg_weight=</span>unweighted_bh_avg, </span>
<span id="cb66-6"><a href="#cb66-6" aria-hidden="true" tabindex="-1"></a>  <span class="at">weight_quantile=</span>unweighted_bh_90_quantile)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>     rejections      avg_weight weight_quantile 
           1973               1               1 </code></pre>
</div>
</div>
<p>E-weighted BH</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb68"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb68-1"><a href="#cb68-1" aria-hidden="true" tabindex="-1"></a>eweighted_bh_rjs <span class="ot">&lt;-</span> <span class="fu">sum</span>(<span class="fu">p.adjust</span>(res<span class="sc">$</span>pvalue <span class="sc">/</span> res<span class="sc">$</span>evalue, <span class="at">method=</span><span class="st">"BH"</span>) <span class="sc">&lt;=</span> alpha_level)</span>
<span id="cb68-2"><a href="#cb68-2" aria-hidden="true" tabindex="-1"></a><span class="fu">stopifnot</span>(eweighted_bh_rjs <span class="sc">==</span> <span class="fu">sum</span>(<span class="fu">ep_BH</span>(res<span class="sc">$</span>pvalue, res<span class="sc">$</span>evalue, alpha_level) ))</span>
<span id="cb68-3"><a href="#cb68-3" aria-hidden="true" tabindex="-1"></a>eweighted_bh_90_quantile <span class="ot">&lt;-</span> <span class="fu">quantile</span>(res<span class="sc">$</span>evalue, <span class="fl">0.9</span>)</span>
<span id="cb68-4"><a href="#cb68-4" aria-hidden="true" tabindex="-1"></a>eweighted_bh_avg <span class="ot">&lt;-</span> <span class="fu">mean</span>(res<span class="sc">$</span>evalue)</span>
<span id="cb68-5"><a href="#cb68-5" aria-hidden="true" tabindex="-1"></a><span class="fu">c</span>(<span class="at">rejections=</span>eweighted_bh_rjs, </span>
<span id="cb68-6"><a href="#cb68-6" aria-hidden="true" tabindex="-1"></a>  <span class="at">avg_weight=</span>eweighted_bh_avg,</span>
<span id="cb68-7"><a href="#cb68-7" aria-hidden="true" tabindex="-1"></a>  <span class="at">weight_quantile=</span>eweighted_bh_90_quantile)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>         rejections          avg_weight weight_quantile.90% 
         2387.00000            17.87776             2.69456 </code></pre>
</div>
</div>
<p>Weighted BH with normalized e-values</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb70"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb70-1"><a href="#cb70-1" aria-hidden="true" tabindex="-1"></a>normalized_weights <span class="ot">&lt;-</span> res<span class="sc">$</span>evalue <span class="sc">/</span> <span class="fu">mean</span>(res<span class="sc">$</span>evalue)</span>
<span id="cb70-2"><a href="#cb70-2" aria-hidden="true" tabindex="-1"></a>normalized_eweighted_bh_rjs <span class="ot">&lt;-</span> <span class="fu">sum</span>(<span class="fu">p.adjust</span>(res<span class="sc">$</span>pvalue <span class="sc">/</span> normalized_weights, <span class="at">method=</span><span class="st">"BH"</span>) <span class="sc">&lt;=</span> alpha_level)</span>
<span id="cb70-3"><a href="#cb70-3" aria-hidden="true" tabindex="-1"></a><span class="fu">stopifnot</span>(normalized_eweighted_bh_rjs <span class="sc">==</span> <span class="fu">sum</span>(<span class="fu">normalized_wBH</span>(res<span class="sc">$</span>pvalue, res<span class="sc">$</span>evalue, alpha_level) ))</span>
<span id="cb70-4"><a href="#cb70-4" aria-hidden="true" tabindex="-1"></a>normalized_eweighted_bh_90_quantile <span class="ot">&lt;-</span> <span class="fu">quantile</span>(normalized_weights, <span class="fl">0.9</span>)</span>
<span id="cb70-5"><a href="#cb70-5" aria-hidden="true" tabindex="-1"></a>normalized_eweighted_bh_avg <span class="ot">&lt;-</span> <span class="dv">1</span></span>
<span id="cb70-6"><a href="#cb70-6" aria-hidden="true" tabindex="-1"></a><span class="fu">c</span>(<span class="at">rejections=</span>normalized_eweighted_bh_rjs, </span>
<span id="cb70-7"><a href="#cb70-7" aria-hidden="true" tabindex="-1"></a>  <span class="at">avg_weight=</span>normalized_eweighted_bh_avg, </span>
<span id="cb70-8"><a href="#cb70-8" aria-hidden="true" tabindex="-1"></a>  <span class="at">weight_quantile=</span>normalized_eweighted_bh_90_quantile)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>         rejections          avg_weight weight_quantile.90% 
       1309.0000000           1.0000000           0.1507214 </code></pre>
</div>
</div>
<p>IHW: apply it in such a way that hypotheses that did not appear in the microarray experiment are treated as a separate categorical group.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb72"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb72-1"><a href="#cb72-1" aria-hidden="true" tabindex="-1"></a>ihw_grouping_non_NA <span class="ot">&lt;-</span> <span class="fu">groups_by_filter</span>(res<span class="sc">$</span>evalue[res<span class="sc">$</span>evalue <span class="sc">!=</span> <span class="dv">1</span>], <span class="dv">14</span>)</span>
<span id="cb72-2"><a href="#cb72-2" aria-hidden="true" tabindex="-1"></a>ihw_grouping <span class="ot">&lt;-</span> <span class="fu">factor</span>(<span class="fu">rep</span>(<span class="dv">0</span>, <span class="fu">nrow</span>(res)), <span class="at">levels=</span><span class="dv">0</span><span class="sc">:</span><span class="dv">14</span>)</span>
<span id="cb72-3"><a href="#cb72-3" aria-hidden="true" tabindex="-1"></a>ihw_grouping[res<span class="sc">$</span>evalue <span class="sc">!=</span> <span class="dv">1</span>] <span class="ot">&lt;-</span> ihw_grouping_non_NA</span>
<span id="cb72-4"><a href="#cb72-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb72-5"><a href="#cb72-5" aria-hidden="true" tabindex="-1"></a>ihw_res <span class="ot">&lt;-</span> <span class="fu">ihw</span>(res<span class="sc">$</span>pvalue, ihw_grouping, <span class="at">alpha=</span>alpha_level)</span>
<span id="cb72-6"><a href="#cb72-6" aria-hidden="true" tabindex="-1"></a>ihw_bh_rjs <span class="ot">&lt;-</span> <span class="fu">rejections</span>(ihw_res)</span>
<span id="cb72-7"><a href="#cb72-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb72-8"><a href="#cb72-8" aria-hidden="true" tabindex="-1"></a>ihw_bh_90_quantile <span class="ot">&lt;-</span> <span class="fu">quantile</span>(<span class="fu">weights</span>(ihw_res), <span class="fl">0.9</span>)</span>
<span id="cb72-9"><a href="#cb72-9" aria-hidden="true" tabindex="-1"></a>ihw_bh_avg <span class="ot">&lt;-</span> <span class="dv">1</span></span>
<span id="cb72-10"><a href="#cb72-10" aria-hidden="true" tabindex="-1"></a><span class="fu">c</span>(<span class="at">rejections=</span>ihw_bh_rjs,</span>
<span id="cb72-11"><a href="#cb72-11" aria-hidden="true" tabindex="-1"></a>  <span class="at">avg_weight=</span>ihw_bh_avg, </span>
<span id="cb72-12"><a href="#cb72-12" aria-hidden="true" tabindex="-1"></a>  <span class="at">weight_quantile=</span>ihw_bh_90_quantile)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>         rejections          avg_weight weight_quantile.90% 
        2012.000000            1.000000            2.110278 </code></pre>
</div>
</div>
<p>Single Index Modulated Multiple Testing</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb74"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb74-1"><a href="#cb74-1" aria-hidden="true" tabindex="-1"></a>sim_mtp_res <span class="ot">&lt;-</span> <span class="fu">sim_mtp</span>(res<span class="sc">$</span>pvalue, res<span class="sc">$</span>microarray_pvalue, alpha_level, <span class="at">Storey=</span><span class="cn">FALSE</span>)</span>
<span id="cb74-2"><a href="#cb74-2" aria-hidden="true" tabindex="-1"></a>sim_rjs <span class="ot">&lt;-</span> <span class="fu">sum</span>(sim_mtp_res)</span>
<span id="cb74-3"><a href="#cb74-3" aria-hidden="true" tabindex="-1"></a><span class="fu">c</span>(<span class="at">rejections=</span>sim_rjs) </span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>rejections 
      2282 </code></pre>
</div>
</div>
</section>
<section id="apply-storey-type-multiple-testing-methods" class="level2">
<h2 class="anchored" data-anchor-id="apply-storey-type-multiple-testing-methods">Apply Storey-type multiple testing methods</h2>
<p>Unweighted</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb76"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb76-1"><a href="#cb76-1" aria-hidden="true" tabindex="-1"></a>unweighted_storey_pi0 <span class="ot">&lt;-</span> <span class="fu">weighted_storey_pi0</span>(res<span class="sc">$</span>pvalue, <span class="dv">1</span>)</span>
<span id="cb76-2"><a href="#cb76-2" aria-hidden="true" tabindex="-1"></a>unweighted_storey_rjs <span class="ot">&lt;-</span> <span class="fu">sum</span>(<span class="fu">tau_weighted_bh</span>(res<span class="sc">$</span>pvalue, <span class="dv">1</span>, <span class="at">Storey=</span><span class="cn">TRUE</span>) <span class="sc">&lt;=</span> alpha_level)</span>
<span id="cb76-3"><a href="#cb76-3" aria-hidden="true" tabindex="-1"></a>unweighted_storey_90_quantile <span class="ot">&lt;-</span> <span class="dv">1</span><span class="sc">/</span>unweighted_storey_pi0</span>
<span id="cb76-4"><a href="#cb76-4" aria-hidden="true" tabindex="-1"></a>unweighted_storey_avg <span class="ot">&lt;-</span> <span class="dv">1</span><span class="sc">/</span>unweighted_storey_pi0</span>
<span id="cb76-5"><a href="#cb76-5" aria-hidden="true" tabindex="-1"></a><span class="fu">c</span>(<span class="at">rejections=</span>unweighted_storey_rjs,</span>
<span id="cb76-6"><a href="#cb76-6" aria-hidden="true" tabindex="-1"></a>  <span class="at">avg_weight=</span>unweighted_storey_avg, </span>
<span id="cb76-7"><a href="#cb76-7" aria-hidden="true" tabindex="-1"></a>  <span class="at">weight_quantile=</span>unweighted_storey_90_quantile)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>     rejections      avg_weight weight_quantile 
    2147.000000        1.344962        1.344962 </code></pre>
</div>
</div>
<p>E-value weights</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb78"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb78-1"><a href="#cb78-1" aria-hidden="true" tabindex="-1"></a>eweighted_storey_pi0 <span class="ot">&lt;-</span> unweighted_storey_pi0</span>
<span id="cb78-2"><a href="#cb78-2" aria-hidden="true" tabindex="-1"></a>eweighted_storey_rjs <span class="ot">&lt;-</span> <span class="fu">sum</span>(<span class="fu">ep_BH</span>(res<span class="sc">$</span>pvalue, res<span class="sc">$</span>evalue, alpha_level, <span class="at">Storey=</span><span class="cn">TRUE</span>))</span>
<span id="cb78-3"><a href="#cb78-3" aria-hidden="true" tabindex="-1"></a>eweighted_storey_90_quantile <span class="ot">&lt;-</span> <span class="fu">quantile</span>(res<span class="sc">$</span>evalue, <span class="fl">0.9</span>) <span class="sc">/</span> eweighted_storey_pi0</span>
<span id="cb78-4"><a href="#cb78-4" aria-hidden="true" tabindex="-1"></a>eweighted_storey_avg <span class="ot">&lt;-</span> <span class="fu">mean</span>(res<span class="sc">$</span>evalue) <span class="sc">/</span> eweighted_storey_pi0</span>
<span id="cb78-5"><a href="#cb78-5" aria-hidden="true" tabindex="-1"></a><span class="fu">c</span>(<span class="at">rejections=</span>eweighted_storey_rjs,</span>
<span id="cb78-6"><a href="#cb78-6" aria-hidden="true" tabindex="-1"></a>  <span class="at">avg_weight=</span>eweighted_storey_avg, </span>
<span id="cb78-7"><a href="#cb78-7" aria-hidden="true" tabindex="-1"></a>  <span class="at">weight_quantile=</span>eweighted_storey_90_quantile)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>         rejections          avg_weight weight_quantile.90% 
         2540.00000            24.04490             3.62408 </code></pre>
</div>
</div>
<p>Weighted Storey with normalized e-values</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb80"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb80-1"><a href="#cb80-1" aria-hidden="true" tabindex="-1"></a>normalized_eweighted_storey_pi0 <span class="ot">&lt;-</span> <span class="fu">weighted_storey_pi0</span>(res<span class="sc">$</span>pvalue, normalized_weights)</span>
<span id="cb80-2"><a href="#cb80-2" aria-hidden="true" tabindex="-1"></a>normalized_eweighted_storey_rjs <span class="ot">&lt;-</span> <span class="fu">sum</span>(<span class="fu">normalized_wBH</span>(res<span class="sc">$</span>pvalue, res<span class="sc">$</span>evalue, alpha_level, <span class="at">Storey=</span><span class="cn">TRUE</span>))</span>
<span id="cb80-3"><a href="#cb80-3" aria-hidden="true" tabindex="-1"></a>normalized_eweighted_storey_90_quantile <span class="ot">&lt;-</span> <span class="fu">quantile</span>(normalized_weights, <span class="fl">0.9</span>) <span class="sc">/</span> normalized_eweighted_storey_pi0</span>
<span id="cb80-4"><a href="#cb80-4" aria-hidden="true" tabindex="-1"></a>normalized_eweighted_storey_avg <span class="ot">&lt;-</span> <span class="dv">1</span> <span class="sc">/</span> normalized_eweighted_storey_pi0</span>
<span id="cb80-5"><a href="#cb80-5" aria-hidden="true" tabindex="-1"></a><span class="fu">c</span>(<span class="at">rejections=</span>normalized_eweighted_storey_rjs,</span>
<span id="cb80-6"><a href="#cb80-6" aria-hidden="true" tabindex="-1"></a>  <span class="at">avg_weight=</span>normalized_eweighted_storey_avg, </span>
<span id="cb80-7"><a href="#cb80-7" aria-hidden="true" tabindex="-1"></a>  <span class="at">weight_quantile=</span>normalized_eweighted_storey_90_quantile)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>         rejections          avg_weight weight_quantile.90% 
       1536.0000000           2.6101615           0.3934072 </code></pre>
</div>
</div>
<p>IHW</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb82"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb82-1"><a href="#cb82-1" aria-hidden="true" tabindex="-1"></a>ihw_storey_res <span class="ot">&lt;-</span> <span class="fu">ihw</span>(res<span class="sc">$</span>pvalue, ihw_grouping, <span class="at">alpha=</span>alpha_level, <span class="at">null_proportion=</span><span class="cn">TRUE</span>)</span>
<span id="cb82-2"><a href="#cb82-2" aria-hidden="true" tabindex="-1"></a>ihw_storey_rjs <span class="ot">&lt;-</span> <span class="fu">rejections</span>(ihw_storey_res)</span>
<span id="cb82-3"><a href="#cb82-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb82-4"><a href="#cb82-4" aria-hidden="true" tabindex="-1"></a>ihw_storey_90_quantile <span class="ot">&lt;-</span> <span class="fu">quantile</span>(<span class="fu">weights</span>(ihw_storey_res), <span class="fl">0.9</span>)</span>
<span id="cb82-5"><a href="#cb82-5" aria-hidden="true" tabindex="-1"></a>ihw_storey_avg <span class="ot">&lt;-</span> <span class="fu">mean</span>(<span class="fu">weights</span>(ihw_storey_res))</span>
<span id="cb82-6"><a href="#cb82-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb82-7"><a href="#cb82-7" aria-hidden="true" tabindex="-1"></a><span class="fu">c</span>(<span class="at">rejections=</span>ihw_storey_rjs,</span>
<span id="cb82-8"><a href="#cb82-8" aria-hidden="true" tabindex="-1"></a>  <span class="at">avg_weight=</span>ihw_storey_avg, </span>
<span id="cb82-9"><a href="#cb82-9" aria-hidden="true" tabindex="-1"></a>  <span class="at">weight_quantile=</span>ihw_storey_90_quantile)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>         rejections          avg_weight weight_quantile.90% 
        2274.000000            1.541129            3.483840 </code></pre>
</div>
</div>
<p>Single Index Modulated Multiple Testing</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb84"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb84-1"><a href="#cb84-1" aria-hidden="true" tabindex="-1"></a>sim_storey_rjs <span class="ot">&lt;-</span> <span class="fu">sum</span>(<span class="fu">sim_mtp</span>(res<span class="sc">$</span>pvalue, res<span class="sc">$</span>microarray_pvalue, alpha_level, <span class="at">Storey=</span><span class="cn">TRUE</span>))</span>
<span id="cb84-2"><a href="#cb84-2" aria-hidden="true" tabindex="-1"></a><span class="fu">c</span>(<span class="at">rejections=</span>sim_storey_rjs) </span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>rejections 
      2479 </code></pre>
</div>
</div>
</section>
<section id="table-of-results" class="level2">
<h2 class="anchored" data-anchor-id="table-of-results">Table of results</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb86"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb86-1"><a href="#cb86-1" aria-hidden="true" tabindex="-1"></a>latex_rownames <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"Unweighted"</span>, </span>
<span id="cb86-2"><a href="#cb86-2" aria-hidden="true" tabindex="-1"></a>                    <span class="st">"E-value Weights"</span>,</span>
<span id="cb86-3"><a href="#cb86-3" aria-hidden="true" tabindex="-1"></a>                    <span class="st">"Normalized E-value Weights"</span>, </span>
<span id="cb86-4"><a href="#cb86-4" aria-hidden="true" tabindex="-1"></a>                    <span class="st">"Indep. Hypothesis Weighting"</span>,</span>
<span id="cb86-5"><a href="#cb86-5" aria-hidden="true" tabindex="-1"></a>                    <span class="st">"Single Index Modulated"</span>)</span>
<span id="cb86-6"><a href="#cb86-6" aria-hidden="true" tabindex="-1"></a>latex_colnames <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"Avg. Weight"</span>, <span class="st">"90</span><span class="sc">\\</span><span class="st">% Weight"</span>, <span class="st">"Discoveries"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb87"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb87-1"><a href="#cb87-1" aria-hidden="true" tabindex="-1"></a>summary_tbl_bh <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(</span>
<span id="cb87-2"><a href="#cb87-2" aria-hidden="true" tabindex="-1"></a>    <span class="at">Average =</span>  <span class="fu">c</span>(unweighted_bh_avg, eweighted_bh_avg, normalized_eweighted_bh_avg, ihw_bh_avg, <span class="cn">NA</span>),</span>
<span id="cb87-3"><a href="#cb87-3" aria-hidden="true" tabindex="-1"></a>    <span class="at">Quantile =</span> <span class="fu">c</span>(unweighted_bh_90_quantile, eweighted_bh_90_quantile, normalized_eweighted_bh_90_quantile, ihw_bh_90_quantile, <span class="cn">NA</span>),</span>
<span id="cb87-4"><a href="#cb87-4" aria-hidden="true" tabindex="-1"></a>    <span class="at">Discoveries =</span> <span class="fu">c</span>(unweighted_bh_rjs, eweighted_bh_rjs, normalized_eweighted_bh_rjs, ihw_bh_rjs, sim_rjs)</span>
<span id="cb87-5"><a href="#cb87-5" aria-hidden="true" tabindex="-1"></a>    )</span>
<span id="cb87-6"><a href="#cb87-6" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(summary_tbl_bh) <span class="ot">&lt;-</span> latex_rownames</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb88"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb88-1"><a href="#cb88-1" aria-hidden="true" tabindex="-1"></a>summary_tbl_storey <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(</span>
<span id="cb88-2"><a href="#cb88-2" aria-hidden="true" tabindex="-1"></a>    <span class="at">Average =</span>  <span class="fu">c</span>(unweighted_storey_avg, eweighted_storey_avg, normalized_eweighted_storey_avg, ihw_storey_avg, <span class="cn">NA</span>),</span>
<span id="cb88-3"><a href="#cb88-3" aria-hidden="true" tabindex="-1"></a>    <span class="at">Quantile =</span> <span class="fu">c</span>(unweighted_storey_90_quantile, eweighted_storey_90_quantile, normalized_eweighted_storey_90_quantile, ihw_storey_90_quantile, <span class="cn">NA</span>),</span>
<span id="cb88-4"><a href="#cb88-4" aria-hidden="true" tabindex="-1"></a>    <span class="at">Discoveries =</span> <span class="fu">c</span>(unweighted_storey_rjs, eweighted_storey_rjs, normalized_eweighted_storey_rjs, ihw_storey_rjs, sim_storey_rjs)</span>
<span id="cb88-5"><a href="#cb88-5" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb88-6"><a href="#cb88-6" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(summary_tbl_storey) <span class="ot">&lt;-</span> latex_rownames</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb89"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb89-1"><a href="#cb89-1" aria-hidden="true" tabindex="-1"></a>summary_tbl <span class="ot">&lt;-</span> <span class="fu">rbind</span>(summary_tbl_bh, summary_tbl_storey)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb90"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb90-1"><a href="#cb90-1" aria-hidden="true" tabindex="-1"></a><span class="fu">kbl</span>(summary_tbl, <span class="at">booktabs =</span> T, <span class="at">digits=</span><span class="dv">2</span>, <span class="at">col.names=</span>latex_colnames, <span class="at">format =</span> <span class="st">'html'</span>, <span class="at">escape =</span> <span class="cn">FALSE</span>) <span class="sc">%&gt;%</span></span>
<span id="cb90-2"><a href="#cb90-2" aria-hidden="true" tabindex="-1"></a>    <span class="fu">pack_rows</span>(<span class="st">"Benjamini-Hochberg"</span>, <span class="dv">1</span>, <span class="dv">5</span>) <span class="sc">%&gt;%</span></span>
<span id="cb90-3"><a href="#cb90-3" aria-hidden="true" tabindex="-1"></a>    <span class="fu">pack_rows</span>(<span class="st">"Storey"</span>, <span class="dv">6</span>, <span class="dv">10</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">

<table>
 <thead>
  <tr>
   <th style="text-align:left;">   </th>
   <th style="text-align:right;"> Avg. Weight </th>
   <th style="text-align:right;"> 90\% Weight </th>
   <th style="text-align:right;"> Discoveries </th>
  </tr>
 </thead>
<tbody>
  <tr grouplength="5"><td colspan="4" style="border-bottom: 1px solid;"><strong>Benjamini-Hochberg</strong></td></tr>
<tr>
   <td style="text-align:left;padding-left: 2em;" indentlevel="1"> Unweighted </td>
   <td style="text-align:right;"> 1.00 </td>
   <td style="text-align:right;"> 1.00 </td>
   <td style="text-align:right;"> 1973 </td>
  </tr>
  <tr>
   <td style="text-align:left;padding-left: 2em;" indentlevel="1"> E-value Weights </td>
   <td style="text-align:right;"> 17.88 </td>
   <td style="text-align:right;"> 2.69 </td>
   <td style="text-align:right;"> 2387 </td>
  </tr>
  <tr>
   <td style="text-align:left;padding-left: 2em;" indentlevel="1"> Normalized E-value Weights </td>
   <td style="text-align:right;"> 1.00 </td>
   <td style="text-align:right;"> 0.15 </td>
   <td style="text-align:right;"> 1309 </td>
  </tr>
  <tr>
   <td style="text-align:left;padding-left: 2em;" indentlevel="1"> Indep. Hypothesis Weighting </td>
   <td style="text-align:right;"> 1.00 </td>
   <td style="text-align:right;"> 2.11 </td>
   <td style="text-align:right;"> 2012 </td>
  </tr>
  <tr>
   <td style="text-align:left;padding-left: 2em;" indentlevel="1"> Single Index Modulated </td>
   <td style="text-align:right;"> NA </td>
   <td style="text-align:right;"> NA </td>
   <td style="text-align:right;"> 2282 </td>
  </tr>
  <tr grouplength="5"><td colspan="4" style="border-bottom: 1px solid;"><strong>Storey</strong></td></tr>
<tr>
   <td style="text-align:left;padding-left: 2em;" indentlevel="1"> Unweighted1 </td>
   <td style="text-align:right;"> 1.34 </td>
   <td style="text-align:right;"> 1.34 </td>
   <td style="text-align:right;"> 2147 </td>
  </tr>
  <tr>
   <td style="text-align:left;padding-left: 2em;" indentlevel="1"> E-value Weights1 </td>
   <td style="text-align:right;"> 24.04 </td>
   <td style="text-align:right;"> 3.62 </td>
   <td style="text-align:right;"> 2540 </td>
  </tr>
  <tr>
   <td style="text-align:left;padding-left: 2em;" indentlevel="1"> Normalized E-value Weights1 </td>
   <td style="text-align:right;"> 2.61 </td>
   <td style="text-align:right;"> 0.39 </td>
   <td style="text-align:right;"> 1536 </td>
  </tr>
  <tr>
   <td style="text-align:left;padding-left: 2em;" indentlevel="1"> Indep. Hypothesis Weighting1 </td>
   <td style="text-align:right;"> 1.54 </td>
   <td style="text-align:right;"> 3.48 </td>
   <td style="text-align:right;"> 2274 </td>
  </tr>
  <tr>
   <td style="text-align:left;padding-left: 2em;" indentlevel="1"> Single Index Modulated1 </td>
   <td style="text-align:right;"> NA </td>
   <td style="text-align:right;"> NA </td>
   <td style="text-align:right;"> 2479 </td>
  </tr>
</tbody>
</table>

</div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb91"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb91-1"><a href="#cb91-1" aria-hidden="true" tabindex="-1"></a><span class="fu">cat</span>(<span class="fu">kbl</span>(summary_tbl, <span class="at">booktabs =</span> T, <span class="at">digits=</span><span class="dv">2</span>, <span class="at">col.names=</span>latex_colnames, <span class="at">format =</span> <span class="st">'latex'</span>, <span class="at">escape =</span> <span class="cn">FALSE</span>) <span class="sc">%&gt;%</span></span>
<span id="cb91-2"><a href="#cb91-2" aria-hidden="true" tabindex="-1"></a>    <span class="fu">pack_rows</span>(<span class="st">"Benjamini-Hochberg"</span>, <span class="dv">1</span>, <span class="dv">5</span>) <span class="sc">%&gt;%</span></span>
<span id="cb91-3"><a href="#cb91-3" aria-hidden="true" tabindex="-1"></a>    <span class="fu">pack_rows</span>(<span class="st">"Storey"</span>, <span class="dv">6</span>, <span class="dv">10</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>
\begin{tabular}[t]{lrrr}
\toprule
  &amp; Avg. Weight &amp; 90\% Weight &amp; Discoveries\\
\midrule
\addlinespace[0.3em]
\multicolumn{4}{l}{\textbf{Benjamini-Hochberg}}\\
\hspace{1em}Unweighted &amp; 1.00 &amp; 1.00 &amp; 1973\\
\hspace{1em}E-value Weights &amp; 17.88 &amp; 2.69 &amp; 2387\\
\hspace{1em}Normalized E-value Weights &amp; 1.00 &amp; 0.15 &amp; 1309\\
\hspace{1em}Indep. Hypothesis Weighting &amp; 1.00 &amp; 2.11 &amp; 2012\\
\hspace{1em}Single Index Modulated &amp; NA &amp; NA &amp; 2282\\
\addlinespace[0.3em]
\multicolumn{4}{l}{\textbf{Storey}}\\
\hspace{1em}Unweighted1 &amp; 1.34 &amp; 1.34 &amp; 2147\\
\hspace{1em}E-value Weights1 &amp; 24.04 &amp; 3.62 &amp; 2540\\
\hspace{1em}Normalized E-value Weights1 &amp; 2.61 &amp; 0.39 &amp; 1536\\
\hspace{1em}Indep. Hypothesis Weighting1 &amp; 1.54 &amp; 3.48 &amp; 2274\\
\hspace{1em}Single Index Modulated1 &amp; NA &amp; NA &amp; 2479\\
\bottomrule
\end{tabular}</code></pre>
</div>
</div>
</section>
</section>
<section id="session-info" class="level1">
<h1>Session Info</h1>
<div class="cell">
<div class="sourceCode cell-code" id="cb93"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb93-1"><a href="#cb93-1" aria-hidden="true" tabindex="-1"></a><span class="fu">session_info</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>─ Session info ───────────────────────────────────────────────────────────────
 setting  value
 version  R version 4.2.1 (2022-06-23)
 os       macOS Catalina 10.15.7
 system   x86_64, darwin17.0
 ui       X11
 language (EN)
 collate  en_US.UTF-8
 ctype    en_US.UTF-8
 tz       America/New_York
 date     2022-11-04
 pandoc   2.19.2 @ /usr/local/bin/ (via rmarkdown)

─ Packages ───────────────────────────────────────────────────────────────────
 package              * version   date (UTC) lib source
 affxparser             1.68.1    2022-04-29 [1] Bioconductor
 affyio                 1.66.0    2022-04-26 [1] Bioconductor
 annotate               1.74.0    2022-04-26 [1] Bioconductor
 AnnotationDbi        * 1.58.0    2022-04-26 [1] Bioconductor
 assertthat             0.2.1     2019-03-21 [1] CRAN (R 4.2.0)
 backports              1.4.1     2021-12-13 [1] CRAN (R 4.2.0)
 Biobase              * 2.56.0    2022-04-26 [1] Bioconductor
 BiocGenerics         * 0.42.0    2022-04-26 [1] Bioconductor
 BiocManager            1.30.18   2022-05-18 [1] CRAN (R 4.2.0)
 BiocParallel           1.30.3    2022-06-05 [1] Bioconductor
 Biostrings           * 2.64.1    2022-08-18 [1] Bioconductor
 bit                    4.0.4     2020-08-04 [1] CRAN (R 4.2.0)
 bit64                  4.0.5     2020-08-30 [1] CRAN (R 4.2.0)
 bitops                 1.0-7     2021-04-24 [1] CRAN (R 4.2.0)
 blob                   1.2.3     2022-04-10 [1] CRAN (R 4.2.0)
 broom                  1.0.1     2022-08-29 [1] CRAN (R 4.2.0)
 cachem                 1.0.6     2021-08-19 [1] CRAN (R 4.2.0)
 cellranger             1.1.0     2016-07-27 [1] CRAN (R 4.2.0)
 cli                    3.4.1     2022-09-23 [1] CRAN (R 4.2.0)
 codetools              0.2-18    2020-11-04 [1] CRAN (R 4.2.1)
 colorspace             2.0-3     2022-02-21 [1] CRAN (R 4.2.0)
 cowplot              * 1.1.1     2020-12-30 [1] CRAN (R 4.2.0)
 crayon                 1.5.2     2022-09-29 [1] CRAN (R 4.2.0)
 curl                   4.3.2     2021-06-23 [1] CRAN (R 4.2.0)
 data.table             1.14.2    2021-09-27 [1] CRAN (R 4.2.0)
 DBI                  * 1.1.3     2022-06-18 [1] CRAN (R 4.2.0)
 dbplyr                 2.2.1     2022-06-27 [1] CRAN (R 4.2.0)
 DelayedArray           0.22.0    2022-04-26 [1] Bioconductor
 DESeq2               * 1.36.0    2022-04-26 [1] Bioconductor
 digest                 0.6.29    2021-12-01 [1] CRAN (R 4.2.0)
 dplyr                * 1.0.10    2022-09-01 [1] CRAN (R 4.2.0)
 ellipsis               0.3.2     2021-04-29 [1] CRAN (R 4.2.0)
 evaluate               0.16      2022-08-09 [1] CRAN (R 4.2.0)
 EValueWeightsPaper   * 0.1.0     2022-11-04 [1] local
 fansi                  1.0.3     2022-03-24 [1] CRAN (R 4.2.0)
 farver                 2.1.1     2022-07-06 [1] CRAN (R 4.2.0)
 fastmap                1.1.0     2021-01-25 [1] CRAN (R 4.2.0)
 fdrtool                1.2.17    2021-11-13 [1] CRAN (R 4.2.0)
 ff                     4.0.7     2022-05-06 [1] CRAN (R 4.2.0)
 forcats              * 0.5.2     2022-08-19 [1] CRAN (R 4.2.0)
 foreach                1.5.2     2022-02-02 [1] CRAN (R 4.2.0)
 fs                     1.5.2     2021-12-08 [1] CRAN (R 4.2.0)
 gargle                 1.2.1     2022-09-08 [1] CRAN (R 4.2.0)
 genefilter             1.78.0    2022-04-26 [1] Bioconductor
 geneplotter            1.74.0    2022-04-26 [1] Bioconductor
 generics               0.1.3     2022-07-05 [1] CRAN (R 4.2.0)
 GenomeInfoDb         * 1.32.4    2022-09-06 [1] Bioconductor
 GenomeInfoDbData       1.2.8     2022-09-20 [1] Bioconductor
 GenomicRanges        * 1.48.0    2022-04-26 [1] Bioconductor
 GEOquery             * 2.64.2    2022-05-15 [1] Bioconductor
 ggplot2              * 3.3.6     2022-05-03 [1] CRAN (R 4.2.0)
 glue                   1.6.2     2022-02-24 [1] CRAN (R 4.2.0)
 googledrive            2.0.0     2021-07-08 [1] CRAN (R 4.2.0)
 googlesheets4          1.0.1     2022-08-13 [1] CRAN (R 4.2.0)
 gtable                 0.3.1     2022-09-01 [1] CRAN (R 4.2.0)
 haven                  2.5.1     2022-08-22 [1] CRAN (R 4.2.0)
 here                 * 1.0.1     2020-12-13 [1] CRAN (R 4.2.0)
 hms                    1.1.2     2022-08-19 [1] CRAN (R 4.2.0)
 htmltools              0.5.3     2022-07-18 [1] CRAN (R 4.2.0)
 htmlwidgets            1.5.4     2021-09-08 [1] CRAN (R 4.2.0)
 httr                   1.4.4     2022-08-17 [1] CRAN (R 4.2.0)
 IHW                  * 1.24.0    2022-04-26 [1] Bioconductor
 IRanges              * 2.30.1    2022-08-18 [1] Bioconductor
 iterators              1.0.14    2022-02-05 [1] CRAN (R 4.2.0)
 jsonlite               1.8.2     2022-10-02 [1] CRAN (R 4.2.0)
 kableExtra           * 1.3.4     2021-02-20 [1] CRAN (R 4.2.0)
 KEGGREST               1.36.3    2022-07-14 [1] Bioconductor
 knitr                  1.40      2022-08-24 [1] CRAN (R 4.2.0)
 labeling               0.4.2     2020-10-20 [1] CRAN (R 4.2.0)
 lattice                0.20-45   2021-09-22 [1] CRAN (R 4.2.1)
 lifecycle              1.0.2     2022-09-09 [1] CRAN (R 4.2.0)
 limma                * 3.52.4    2022-09-27 [1] Bioconductor
 locfit                 1.5-9.6   2022-07-11 [1] CRAN (R 4.2.0)
 lpsymphony             1.24.0    2022-04-26 [1] Bioconductor (R 4.2.0)
 lubridate              1.8.0     2021-10-07 [1] CRAN (R 4.2.0)
 magrittr               2.0.3     2022-03-30 [1] CRAN (R 4.2.0)
 Matrix                 1.5-1     2022-09-13 [1] CRAN (R 4.2.0)
 MatrixGenerics       * 1.8.1     2022-06-30 [1] Bioconductor
 matrixStats          * 0.62.0    2022-04-19 [1] CRAN (R 4.2.0)
 memoise                2.0.1     2021-11-26 [1] CRAN (R 4.2.0)
 modelr                 0.1.9     2022-08-19 [1] CRAN (R 4.2.0)
 mouse4302.db         * 3.13.0    2022-09-30 [1] Bioconductor
 munsell                0.5.0     2018-06-12 [1] CRAN (R 4.2.0)
 oligo                * 1.60.0    2022-04-26 [1] Bioconductor
 oligoClasses         * 1.58.0    2022-04-26 [1] Bioconductor
 org.Mm.eg.db         * 3.15.0    2022-09-30 [1] Bioconductor
 pd.mouse430.2        * 3.12.0    2022-10-05 [1] Bioconductor
 pillar                 1.8.1     2022-08-19 [1] CRAN (R 4.2.0)
 pkgconfig              2.0.3     2019-09-22 [1] CRAN (R 4.2.0)
 png                    0.1-7     2013-12-03 [1] CRAN (R 4.2.0)
 preprocessCore         1.58.0    2022-04-26 [1] Bioconductor
 purrr                * 0.3.4     2020-04-17 [1] CRAN (R 4.2.0)
 R.methodsS3            1.8.2     2022-06-13 [1] CRAN (R 4.2.0)
 R.oo                   1.25.0    2022-06-12 [1] CRAN (R 4.2.0)
 R.utils                2.12.0    2022-06-28 [1] CRAN (R 4.2.0)
 R6                     2.5.1     2021-08-19 [1] CRAN (R 4.2.0)
 RColorBrewer           1.1-3     2022-04-03 [1] CRAN (R 4.2.0)
 Rcpp                   1.0.9     2022-07-08 [1] CRAN (R 4.2.0)
 RCurl                  1.98-1.9  2022-10-03 [1] CRAN (R 4.2.1)
 readr                * 2.1.3     2022-10-01 [1] CRAN (R 4.2.0)
 readxl                 1.4.1     2022-08-17 [1] CRAN (R 4.2.0)
 reprex                 2.0.2     2022-08-17 [1] CRAN (R 4.2.0)
 rlang                  1.0.6     2022-09-24 [1] CRAN (R 4.2.0)
 rmarkdown              2.16      2022-08-24 [1] CRAN (R 4.2.0)
 rprojroot              2.0.3     2022-04-02 [1] CRAN (R 4.2.0)
 RSQLite              * 2.2.18    2022-10-04 [1] CRAN (R 4.2.1)
 rstudioapi             0.14      2022-08-22 [1] CRAN (R 4.2.0)
 rvest                  1.0.3     2022-08-19 [1] CRAN (R 4.2.0)
 S4Vectors            * 0.34.0    2022-04-26 [1] Bioconductor
 scales                 1.2.1     2022-08-20 [1] CRAN (R 4.2.0)
 sessioninfo          * 1.2.2     2021-12-06 [1] CRAN (R 4.2.0)
 slam                   0.1-50    2022-01-08 [1] CRAN (R 4.2.0)
 stringi                1.7.8     2022-07-11 [1] CRAN (R 4.2.0)
 stringr              * 1.4.1     2022-08-20 [1] CRAN (R 4.2.0)
 SummarizedExperiment * 1.26.1    2022-05-01 [1] Bioconductor
 survival               3.4-0     2022-08-09 [1] CRAN (R 4.2.0)
 svglite                2.1.0     2022-02-03 [1] CRAN (R 4.2.0)
 systemfonts            1.0.4     2022-02-11 [1] CRAN (R 4.2.0)
 tibble               * 3.1.8     2022-07-22 [1] CRAN (R 4.2.0)
 tidyr                * 1.2.1     2022-09-08 [1] CRAN (R 4.2.0)
 tidyselect             1.1.2     2022-02-21 [1] CRAN (R 4.2.0)
 tidyverse            * 1.3.2     2022-07-18 [1] CRAN (R 4.2.0)
 tzdb                   0.3.0     2022-03-28 [1] CRAN (R 4.2.0)
 utf8                   1.2.2     2021-07-24 [1] CRAN (R 4.2.0)
 vctrs                  0.4.2     2022-09-29 [1] CRAN (R 4.2.0)
 viridisLite            0.4.1     2022-08-22 [1] CRAN (R 4.2.0)
 webshot                0.5.4     2022-09-26 [1] CRAN (R 4.2.0)
 withr                  2.5.0     2022-03-03 [1] CRAN (R 4.2.0)
 xfun                   0.33      2022-09-12 [1] CRAN (R 4.2.0)
 XML                    3.99-0.10 2022-06-09 [1] CRAN (R 4.2.0)
 xml2                   1.3.3     2021-11-30 [1] CRAN (R 4.2.0)
 xtable                 1.8-4     2019-04-21 [1] CRAN (R 4.2.0)
 XVector              * 0.36.0    2022-04-26 [1] Bioconductor
 yaml                   2.3.5     2022-02-21 [1] CRAN (R 4.2.0)
 zlibbioc               1.42.0    2022-04-26 [1] Bioconductor

 [1] /Library/Frameworks/R.framework/Versions/4.2/Resources/library

──────────────────────────────────────────────────────────────────────────────</code></pre>
</div>
</div>
</section>

</main>
<!-- /main column -->
<script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const icon = "";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const clipboard = new window.ClipboardJS('.code-copy-button', {
    target: function(trigger) {
      return trigger.previousElementSibling;
    }
  });
  clipboard.on('success', function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    setTimeout(function() {
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  });
  function tippyHover(el, contentFn) {
    const config = {
      allowHTML: true,
      content: contentFn,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start'
    };
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      // use id or data attribute instead here
      let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      return note.innerHTML;
    });
  }
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const cites = ref.parentNode.getAttribute('data-cites').split(' ');
    tippyHover(ref, function() {
      var popup = window.document.createElement('div');
      cites.forEach(function(cite) {
        var citeDiv = window.document.createElement('div');
        citeDiv.classList.add('hanging-indent');
        citeDiv.classList.add('csl-entry');
        var biblioDiv = window.document.getElementById('ref-' + cite);
        if (biblioDiv) {
          citeDiv.innerHTML = biblioDiv.innerHTML;
        }
        popup.appendChild(citeDiv);
      });
      return popup.innerHTML;
    });
  }
});
</script>
</div> <!-- /content -->



</body></html>